src/filesystem.cpp

```diff
@@ -1129,28 +1129,30 @@ namespace otto::filesystem {
                        std::error_code& ec)
   {
     _entries = std::make_shared<std::vector<DEntr>>();
-    DIR* dir;
-    dirent* de;
 
     bool skip_denied = (options & directory_options::skip_permission_denied) !=
                        directory_options::none;
 
-    errno = 0;
-    dir   = opendir(p.c_str());
+    auto dir = opendir(p.c_str());
```

Comment:
i generally prefer leaving the `*` marker on `auto`. Not something im very consistent with myself though.

Reply:
I'm not sure what you're referring to here?

Reply:
`auto* dir = opendir(p.c_str());` - really not very important, just an observation i made while reading

Reply:
You mean changing that line to

```c++
auto* dir = opendir(p.c_str());
```